WordPress/WordPressTest/GutenbergSettingsTests.swift

```diff
@@ -113,4 +119,107 @@ class GutenbergSettingsTests: XCTestCase {
     func testWPComAccountsDefaultsToAztec() {
         XCTAssertFalse(mustUseGutenberg)
     }
+
+    // MARK: - Tests for Autoenabling gutenberg
+
+    // Autoenable on new installs
+
+    func testDoNotAutoenableIfUsersSwitchesToGutenberg() {
+        settings.setGutenbergEnabled(true, for: blog)
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertFalse(shouldAutoenableGutenberg)
+    }
+
+    func testDoNotAutoenableIfUsersSwitchesToGutenbergAndBackToAztec() {
+        settings.setGutenbergEnabled(true, for: blog)
+        settings.setGutenbergEnabled(false, for: blog)
+        post.content = gutenbergContent
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertFalse(shouldAutoenableGutenberg)
+    }
+
+    func testAutoenableWhenSetToAztecOpeningGutenbergPost() {
+        post.content = gutenbergContent
+        blog.mobileEditor = .aztec
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertTrue(shouldAutoenableGutenberg)
+    }
+
+    // Autoenable on new blogs
+
+    func testAutoenableOnNewPostAndNewBlogs() {
+        settings.softSetGutenbergEnabled(true, for: blog)
+        settings.setGutenbergEnabled(true, for: blog) // Called after sync
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertTrue(shouldAutoenableGutenberg)
+    }
+
+    func testAutoenableOnExistingPostAndNewBlogs() {
+        settings.softSetGutenbergEnabled(true, for: blog)
+        settings.setGutenbergEnabled(true, for: blog) // Called after sync
+
+        post.content = gutenbergContent
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertTrue(shouldAutoenableGutenberg)
+    }
+
+    func testAutoenableOnNewBlogsOccoursOnlyOnce() {
+        settings.softSetGutenbergEnabled(true, for: blog)
+        settings.setGutenbergEnabled(true, for: blog) // Called after sync
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertTrue(shouldAutoenableGutenberg)
+
+        settings.willShowDialog(for: blog) // Called by EditorFactory
+
+        XCTAssertFalse(shouldAutoenableGutenberg)
+    }
+
+    // Autoenable after Migration
+
+    func testDoNotAutoenableAfterMigrationSetToGutenberg() {
+        database.set(true, forKey: GutenbergSettings.Key.appWideEnabled)
+        migration()
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertFalse(shouldAutoenableGutenberg)
+    }
+
+    func testDoNotAutoenableAfterMigrationSetToAztec() {
+        database.set(false, forKey: GutenbergSettings.Key.appWideEnabled)
+        migration()
+
+        post.content = gutenbergContent
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertFalse(shouldAutoenableGutenberg)
+    }
+
+    func testAutoenableAfterMigrationNotSet() {
+        database.set(nil, forKey: GutenbergSettings.Key.appWideEnabled)
+        migration()
+
+        post.content = gutenbergContent
+
+        XCTAssertTrue(mustUseGutenberg)
+        XCTAssertTrue(shouldAutoenableGutenberg)
+    }
+
+    func migration() {
```

Comment:
These tests are probably better suited to be run with the actual migration test, but we need to inject an UserDefaults double in the migration itself first.